pkg/apis/lighthouse/v1alpha1/types.go

```diff
@@ -393,3 +412,10 @@ func (a *ActivityRecord) RunningStages() []string {
 	}
 	return running
 }
+
+// JenkinsSpec is optional parameters for Jenkins jobs.
+// Currently, the only parameter supported is for telling
+// jenkins-operator that the job is generated by the https://go.cloudbees.com/docs/plugins/github-branch-source/#github-branch-source plugin
+type JenkinsSpec struct {
+	GitHubBranchSourceJob bool `json:"github_branch_source_job,omitempty"`
```

Comment:
@abayer - that's GitHub specific really. Apparently for PR builds the code relies on the Git Branch Source plugin. In the long run we need something else here (provider independent), the question is whether we should have this interim?

Reply:
Rename it to `BranchSourceJob` - the mechanics of Jenkins branch source are the same from our perspective, so let's keep it around.

Reply:
done